app/service/planning_svc.py

```diff
@@ -119,6 +119,8 @@ async def _check_and_generate_cleanup_links(self, agent, operation):
     async def _generate_new_links(self, operation, agent, abilities, link_status):
         links = []
         for a in await agent.capabilities(abilities):
+            if a.code and a.HOOKS:
```

Comment:
do we need a.code as a requirement to execute the hooks on an ability here?

Reply:
probably not - it should be find to just check for hooks